torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -661,9 +665,24 @@ struct NCCLTraceBuffer {
       entries.push_back(dict);
     }
 
+    // convert ncclDumpMap into a dictionary
+    auto per_comm_dict = new_dict();
```

Comment:
@zdevito does IValue provide infra to recursively wrap a c++ map of compatible stuff into an IValue or do we have to do the for loops here?